crates/llm-base/src/inference_session.rs

```diff
@@ -335,6 +335,34 @@ impl InferenceSession {
         Ok(())
     }
 
+    /// Removes `num` tokens from the end of the buffer. Roughly the inverse of `feed_prompt`.
+    pub fn delete_tokens(
+        &mut self,
+        model: &dyn Model,
+        num: usize,
+    ) -> Result<Vec<u8>, InferenceError> {
+        assert!(
+            num < self.n_past,
+            "Cannot remove starting token",
+        );
```

Comment:
We probably need some better error handling here.

Reply:
Good point, I'll see how the other methods do it and try my hand at it. Thanks for the quick feedback!